src/routes/Users.js

```diff
@@ -135,6 +136,40 @@ export default class Users extends API {
     return Users.presenter.render(result.rows, API.meta(result, userQuery))
   }
 
+  @DELETE('/users/:id')
+  @websocket('users', 'delete')
+  @required('password')
+  async delete (ctx) {
+    let user = await User.findOne({
+      where: {
+        id: ctx.params.id
+      }
+    })
+
+    if (!user) {
+      throw new NotFoundAPIError({parameter: 'id'})
+    }
+
+    this.requireWritePermission({connection: ctx, entity: user})
+
+    let isAuthenticated = await Authentication.passwordAuthenticate({email: user.email, password: ctx.data.password})
+    if (!isAuthenticated) {
+      throw new UnauthorizedAPIError({pointer: '/data/attributes/password'})
+    }
+
+    let rats = await Rat.findAll({
+      where: {
+        userId: ctx.params.id
+      }
+    })
+
+    rats.forEach(rat => rat.destroy())
```

Comment:
Should these destroy operations be awaited?
something like
`await Promise.all(rats.forEach(rat => rat.destroy()))`